File: frontend/src/shared/types/actividad.ts
/**
 * Types for Actividad (Activity/Audit) domain
 *
 * Ref: docs/specs/15-ACTIVIDAD-RECIENTE-AUDITORIA.md
 */

/**
 * Tipos de actividad en el sistema.
 * Mirrors: backend/app/models/enums.py::TipoActividadEnum
 */
export type TipoActividad =
  | 'USUARIO_CREADO'
  | 'MATERIA_CREADA'
  | 'COMISION_CREADA'
  | 'RUBRICA_CREADA';

/**
 * TipoActividad values object for runtime usage.
 */
export const TIPO_ACTIVIDAD = {
  USUARIO_CREADO: 'USUARIO_CREADO' as const,
  MATERIA_CREADA: 'MATERIA_CREADA' as const,
  COMISION_CREADA: 'COMISION_CREADA' as const,
  RUBRICA_CREADA: 'RUBRICA_CREADA' as const,
} as const;

export interface Actividad {
  id: number;
  tipo: TipoActividad;
  descripcion: string;
  entidad_id: number;
  entidad_nombre: string;
  usuario_id: number | null;
  created_at: string;
  usuario_nombre: string | null;
  usuario_rol: string | null;
}

export interface ActividadListResponse {
  items: Actividad[];
  total: number;
}

File: backend/app/schemas/actividad.py
# app/schemas/actividad.py
"""
Pydantic schemas for Actividad (Activity/Audit) model.

Ref: docs/specs/15-ACTIVIDAD-RECIENTE-AUDITORIA.md
"""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import TipoActividadEnum


class ActividadBase(BaseModel):
    """Base schema para Actividad."""

    tipo: TipoActividadEnum
    descripcion: str = Field(..., max_length=500)
    entidad_id: int
    entidad_nombre: str = Field(..., max_length=255)


class ActividadCreate(ActividadBase):
    """Schema para crear una actividad."""

    usuario_id: int | None = None
    metadatos: str | None = None


class ActividadResponse(ActividadBase):
    """Schema para respuesta de actividad."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    usuario_id: int | None
    created_at: datetime

    # Datos del usuario que realizó la acción (opcional)
    usuario_nombre: str | None = None
    usuario_rol: str | None = None


class ActividadListResponse(BaseModel):
    """Schema para lista de actividades."""

    items: list[ActividadResponse]
    total: int
